spec/api-shell-spec.js

```diff
@@ -79,4 +79,48 @@ describe('shell module', () => {
       assert.deepEqual(shell.readShortcutLink(tmpShortcut), change)
     })
   })
+
+  describe('shell.moveItemToTrash', () => {
+    const filePath = path.join(os.tmpdir(), `${Date.now()}.md`)
+    fs.writeFileSync(filePath, '# Hello')
+
+    it('moves file to trash asynchronously', done => {
+      shell.moveItemToTrash(filePath).then(() => {
+        assert.equal(fs.existsSync(filePath), false)
+        done()
+      }).catch(e => {
+        done(e)
+      })
+      assert.equal(fs.existsSync(filePath), true)
+    })
+
+    it('rejects the promise in case of error', done => {
+      shell.moveItemToTrash(filePath + '_does_not_exist').then(() => {
+        done(new Error('Should be unreachable'))
+      }).catch(e => {
+        assert.ok(e)
+        done(e)
```

Comment:
It can be tested easier with [chai-as-promised](https://github.com/domenic/chai-as-promised):
```js
it('gets rejected', () => {
  return expect(shell.moveItemToTrash(nonexistentFilePath))
      .to.be.eventually.rejectedWith(Error)
})
```

Check [api-app-spec.js](https://github.com/electron/electron/blob/fcc82eb/spec/api-app-spec.js) to find out how to get that stuff available for your tests.